### DISCUSSION TITLE
Using @auth/sveltekit with @next-auth/prisma-adapter?

### DISCUSSION THREAD

**Discussion by KoljaL** [nextauthjs/next-auth#6074] [Help]
How can i use `@auth/sveltekit ` with Prisma?

I found this doc: https://authjs.dev/reference/adapters/prisma
but it is not SvelteKit specific.

Thanks a lot for making `authJS` framework-agnostic :-)

**Comment by KoljaL** [comment]
I tried the following:

Follow this: https://authjs.dev/reference/adapters/prisma

And did not install `next-auth`, but `@auth/core @auth/sveltekit` instead.
(Maybe this is the problem?)


created my `hooks.server.ts`:
```javascript
import SvelteKitAuth from "@auth/sveltekit"
import GitHub from "@auth/core/providers/github"
import { GITHUB_ID, GITHUB_SECRET } from "$env/static/private"
import { PrismaAdapter } from "@next-auth/prisma-adapter"
import { PrismaClient } from "@prisma/client"

const prisma = new PrismaClient()
 
export const handle = SvelteKitAuth({
  adapter: PrismaAdapter(prisma),
  providers: [GitHub({ clientId: GITHUB_ID, clientSecret: GITHUB_SECRET })],
})
```
Did the rest stuff for Prisma (schema, migrate, generate)

After `npm run dev` try to Sign in with GitHub creates a redirect to: `http://localhost:5173/auth/signin?error=Callback`
And an error in the terminal:

```bash
16:38:06 [vite] page reload src/hooks.server.ts
[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR]
https://next-auth.js.org/errors#oauth_callback_handler_error Value of "this" must be of type Crypto {
  message: 'Value of "this" must be of type Crypto',
  stack: 'TypeError [ERR_INVALID_THIS]: Value of "this" must be of type Crypto\n' +
    '    at new NodeError (node:internal/errors:393:5)\n' +
    '    at randomUUID (node:internal/crypto/webcrypto:80:30)\n' +
    '    at callbackHandler (file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@auth+core@0.1.4/node_modules/@auth/core/lib/callback-handler.js:109:35)\n' +
    '    at async Module.callback (file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@auth+core@0.1.4/node_modules/@auth/core/lib/routes/callback.js:73:54)\n' +
    '    at async AuthHandlerInternal (file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@auth+core@0.1.4/node_modules/@auth/core/index.js:86:38)\n' +
    '    at async Proxy.AuthHandler (file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@auth+core@0.1.4/node_modules/@auth/core/index.js:216:30)\n' +
    '    at async respond (file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@sveltejs+kit@1.0.0_svelte@3.55.0+vite@4.0.1/node_modules/@sveltejs/kit/src/runtime/server/index.js:229:20)\n' +
    '    at async file:///Users/xxx/Code/SK-Auth-Variants/SK-AuthJS/node_modules/.pnpm/@sveltejs+kit@1.0.0_svelte@3.55.0+vite@4.0.1/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:444:22',
  name: 'TypeError'
}
```
In the database (SQLite), the table `User`and `Account` get a correct looking entry.

Without the line: `adapter: PrismaAdapter(prisma),` it's works like a charm :-)

**Reply by JonathonRP** [reply]
do you mean prisma works without the adapter? as in without that line the users still get entry in to prisma database?

**Comment by osdiab** [comment]
I'm guessing this isn't ready yet, but more generally I was wondering if the next-auth providers are compatible with the new generalized `@auth/core` packages or not yet. Would be awesome to use this in my next project, but I'm considering Sveltekit and that would make it tougher!

**Comment by califlower** [comment]
The solution to this is simple. Until they fix it just define a new `generateSessionToken` function in your auth options like so 

```
	session: {
		// Choose how you want to save the user session.
		// The default is `"jwt"`, an encrypted JWT (JWE) stored in the session cookie.
		// If you use an `adapter` however, we default it to `"database"` instead.
		// You can still force a JWT session by explicitly defining `"jwt"`.
		// When using `"database"`, the session cookie will only contain a `sessionToken` value,
		// which is used to look up the session in the database.
		strategy: 'database',

		// Seconds - How long until an idle session expires and is no longer valid.
		maxAge: 30 * 24 * 60 * 60, // 30 days

		// Seconds - Throttle how frequently to write to database to extend a session.
		// Use it to limit write operations. Set to 0 to always update the database.
		// Note: This option is ignored if using JSON Web Tokens
		updateAge: 24 * 60 * 60, // 24 hours

		// The session token is usually either a random UUID or string, however if you
		// need a more customized session token string, you can define your own generate function.
		generateSessionToken: () => {
			return crypto.randomUUID();
		}
	}
```

**Reply by RonB** [reply]
I needed to write a special database adapter because i am using surrealdb, but it looks like i am getting somewhere. Also thank you @califlower because I got also stuck with the default
generateSessionToken function which depends on randomUUID() which is a function crypto.

**Reply by Pab450** [reply]
thx for ur help @califlower

**Comment by joshnuss** [comment]
It appears to be fixed in latest versions.

Tested with:

```diff
+"@auth/sveltekit": "^0.2.2",
+"@next-auth/prisma-adapter": "^1.0.6",
```

**Comment by Ladvace** [comment]
I'm still getting this error with `@auth/solid-start`,  prima adapter, and GitHub adapter

**Reply by zimonitrome** [reply]
Me too... I started using `@solid-auth/base` and now tried to replace it with `@auth/solid-start`. I'm unsure which one to use...

**Reply by Ladvace** [reply]
are you using the github adapter or another one?

**Reply by zimonitrome** [reply]
Sorry, I'm using the Google and Github *provider* and only ther Prisma adapter.

**Comment by JonathonRP** [comment]
Fyi there is now an @auth/prisma-adapter and would recommend using that over next-auth/prisma-adapter.

**Reply by Ladvace** [reply]
I just tried this instead of the next-auth one but I still get the same error

**Reply by JonathonRP** [reply]
Have you tried setting the db strategy?
This is what I did and I never had a problem.
